### PR TITLE
[PUI] Add coverage testing

### PR DESCRIPTION
**Pull request by matmair**
This adds code coverage for the PUI tests and reporting to coveralls.

### CODE REVIEW

**Comment by matmair** [comment]
@wolflu05 you are more experienced in typescript/frontend. Is anything sticking out with this implementation?

**Comment by wolflu05** [comment]
Sorry, I have not that much experience with frontend testing, never really done this before.

Regarding the changes on the vite config, does this mean we now ship sourcemaps with all builds? Is that wanted?

And the plugin you added, has that any influence on the build or dev environment?

**Comment by matmair** [comment]
I have removed tests I added before, they lowered the coverage a lot and were not that great - still wrapping my head around writing efficient tests. The focus of this PR - coverage - works great, expanding the tests should probably be part of further PUI development.

File: .github/workflows/qc_checks.yaml [hunk 1]
```diff
@@ -487,13 +488,30 @@ jobs:
...
-        run: cd src/frontend &&  npx playwright test
+        run: cd src/frontend && npx nyc playwright test
       - uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # pin@v4.3.1
+        name: Upload playwright report
         if: always()
         with:
           name: playwright-report
           path: src/frontend/playwright-report/
           retention-days: 30
+      - name: Report coverate
```
Comment (wolflu05) [inline .github/workflows/qc_checks.yaml:499]: typo
```suggestion
      - name: Report coverage
```

Comment (matmair) [reply .github/workflows/qc_checks.yaml]: Thank you!

**Comment by SchrodingersGat** [comment]
@matmair please address conflicts here

**Comment by matmair** [comment]
@SchrodingersGat merge is in!

**Pull request merged**